nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,503 @@
+          {
+            "org.ice4j.ice.harvest.NAT_HARVESTER_LOCAL_ADDRESS" = "192.168.1.42";
+            "org.ice4j.ice.harvest.NAT_HARVESTER_PUBLIC_ADDRESS" = "1.2.3.4";
+          }
+        '';
+        description = ''
+          Additional Java properties passed to jitsi-videobridge.
+
+          The following extra properties need to be added when running behind NAT:
+
+          <programlisting>
+          org.ice4j.ice.harvest.NAT_HARVESTER_LOCAL_ADDRESS = <replaceable>Local.IP.Address</replaceable>
+          org.ice4j.ice.harvest.NAT_HARVESTER_PUBLIC_ADDRESS = <replaceable>Public.IP.Address</replaceable>
+          </programlisting>
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = bool;
+        default = false;
+        description = ''
+          Whether to open ports in the firewall for the videobridge.
+        '';
+      };
+    };
+
+    jicofo = {
+      config = mkOption {
+        type = attrsOf str;
+        default = { };
+        example = literalExample ''
+          {
+            "org.jitsi.jicofo.ALWAYS_TRUST_MODE_ENABLED" = "true";
+            "org.jitsi.jicofo.auth.URL" = "XMPP:jitsi-meet.example.com";
+          }
+        '';
+        description = ''
+          Contents of the <filename>sip-communicator.properties</filename> configuration file for jicofo.
+        '';
+      };
+    };
+
+    nginx.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to enable nginx virtual host that will serve the javascript application and act as
+        a proxy for the XMPP server. Further nginx configuration can be done by adapting
+        <option>services.nginx.virtualHosts.&lt;hostName&gt;</option>. It is highly recommended to
+        enable the <option>enableACME</option> and <option>forceSSL</option> options:
+
+        <programlisting>
+        services.nginx.virtualHosts.''${config.services.jitsi-meet.hostName} = {
+          enableACME = true;
+          forceSSL = true;
+        };
+        </programlisting>
+      '';
+    };
+
+    prosody.enable = mkOption {
+      type = bool;
+      default = true;
+      description = ''
+        Whether to configure Prosody to relay XMPP messages between Jitsi Meet components. Turn this
+        off if you want to configure it manually.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.jitsi-meet.jicofo.config = mapAttrs (_: v: mkDefault v) {
+      "org.jitsi.jicofo.BRIDGE_MUC" = "jvbbrewery@internal.${cfg.hostName}";
+    };
+
+    services.prosody = mkIf cfg.prosody.enable {
+      enable = mkDefault true;
+      modules = {
+        admin_adhoc = mkDefault false;
+        bosh = mkDefault true;
+        ping = mkDefault true;
+        roster = mkDefault true;
+        saslauth = mkDefault true;
+        tls = mkDefault true;
+      };
+      extraModules = [ "pubsub" ];
+      extraConfig = ''
+        certificates = "${config.services.prosody.dataDir}"
```

Comment:
This makes it pretty hard to use acme certificates here, as we hardcode the location to dataDir.

IMHO, these things should be made more configurable in the prosody module, before using it here.